execution_engine/src/storage/protocol_data.rs

```diff
@@ -1,17 +1,16 @@
-use crate::shared::wasm_costs::{WasmCosts, WASM_COSTS_SERIALIZED_LENGTH};
+use crate::shared::wasm_config::WasmConfig;
 use casper_types::{
     bytesrepr::{self, FromBytes, ToBytes},
-    ContractHash, HashAddr, KEY_HASH_LENGTH,
+    ContractHash, HashAddr,
 };
 use std::collections::BTreeMap;
 
-const PROTOCOL_DATA_SERIALIZED_LENGTH: usize = WASM_COSTS_SERIALIZED_LENGTH + 4 * KEY_HASH_LENGTH;
 const DEFAULT_ADDRESS: [u8; 32] = [0; 32];
 
 /// Represents a protocol's data. Intended to be associated with a given protocol version.
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
I'm curious why you opted to not make WasmConfig Copy, which forced you to remove it here and add a lot of .clone()'s?

Reply:
Per slack convo with Michal, it's due to a vec used in chainspec values that hold function arguments. We discussed using a Tuple instead if time allows.